### ISSUE TITLE
weight prediction baselines

### ISSUE THREAD

**Issue by yuchenhou** [yuchenhou/elephant#34]
### paper

[Prediction of Links and Weights in Networks by Reliable Routes](http://www.nature.com/articles/srep12261?WT.ec_id=SREP-639-20150728&spMailingID=49194604&spUserID=ODkwMTM2NjQzMgS2&spJobID=723669330&spReportId=NzIzNjY5MzMwS0)
### Introduction

For a weighted undirected graph where links with higher weights represent stronger relations between nodes (e.g., interactions between proteins, genes, authors, neurons), weight prediction is an important task.
### problem
- given: the weights of links in a training set
- task: predict the weights of links in the test set to minimize RMSE

### solution
1. for each node pair in the entire data set, calculate the similarity of the pair (9 variants of similarity evaluated in experiments)
2. let a linear regression model `weight = F(similarity) = k * similarity` learn on the training set
3. let model F predict on the test set

### comment
- The description of the learning (or testing) process in the paper (the paragraphs around equation 3) seems wrong, because the model learns on the test set. This is a not a correct learning process or testing process. I didn't find a correct description in the paper so I made one up.
- Each of the 9 variants of the pair similarity is a sum of "strengths" of all 2-link paths connecting the pair. The "strength" of a path is a sum or product of weights of links in the path, normalized by some factor. However, it is unclear why the paper only considers 2-link paths, but not 1-link, 3-link, 4-link ones.
- The paper states this approach does not work well for graphs where higher weights represent weaker relations. Is this because dissimilarity is not additive?

**Comment by yuchenhou** [comment]
I followed the paper's references to related work on weighted graph, but findings were little. This one paper is on weight prediction: http://comnet.oxfordjournals.org/content/early/2014/06/26/comnet.cnu026.short
but unfortunately our school doesn't have subscription for it. All the other papers are not on weight prediction. Will expand my search later.

**Comment by lbholder** [comment]
Looks like an early version of the paper is available at http://arxiv.org/abs/1404.0431.

**Comment by yuchenhou** [comment]
### paper
[Learning Latent Block Structure in Weighted Networks](http://arxiv.org/pdf/1404.0431v2.pdf)

### Introduction
A weighted directed heterogeneous graph can have a 2-level hierarchy:
- upper level: the heterogeneous graph is a set of homogeneous blocks connected by bundles
- lower level: a homogeneous block is a set of nodes with similar connections connected by links

### Solution
- the model contains parameters including membership (the block each node is in) and bundles' weights
- input: a pair of nodes
- output: the weight of the link connecting the pair of nodes

### Comments
- too much statistics, didn't read all
- this solution has a side effect of clustering nodes into blocks by assigning membership, a useful thing by itsefl

**Comment by yuchenhou** [comment]
found the implementation of [WSBM](http://tuvalu.santafe.edu/~aaronc/wsbm/)
need to figure out how to run it

**Comment by yuchenhou** [comment]
## paper
[Weight prediction in complex networks based on neighbor set](https://www.nature.com/articles/srep38080?WT.ec_id=SREP-639-20161206&spMailingID=52927796&spUserID=ODkwMTM2NjQzMgS2&spJobID=1061117970&spReportId=MTA2MTExNzk3MAS2)

## problem
Weight link prediction, same as above.

## solution
Not a machine learning approach, very hard to understand. The assumption it is based on is `Our method relies on the assumption that the formation of link weights is regulated by local clusterings in which homogenous links tend to have similar weights.` also very hard to understand.
